### PR TITLE
chore!: Normalize repository, dropping node <10.13 support

### LINKED ISSUE #49: Future: Update chokidar
Hi,

Could you please update `chokidar` to 3.x. It's been a couple of years and it seems that `fsevents` used by `chokidar` 2.x is broken when compiled against the newer Node APIs (v12.x). However that has to be fixed in newer `chokidar` and `fsevents`.

### LINKED ISSUE #55: Upgrade chokidar for real. v2 will break on node 14
Hey @phated and folks,

We've released Chokidar 3 one year ago. See my post on that: [Chokidar 3: How to save 32TB of traffic every week with one NPM package](https://paulmillr.com/posts/chokidar-3-save-32tb-of-traffic/)

The changes are pretty big. First of all, the package is 16 times smaller and uses 15 dependencies instead of 201. We've switched to n-api, which means users won't download fsevents binaries anymore. n-api is included in every nodejs installation, and it's tiny. Directory walking has been improved massively with stream implementation of readdirp. Massive RAM & CPU improvements etc.

*I saw previous issues* (#41 #42 #47 #48 #49 etc), but unfortunately they're either closed or closed for comments. I think it's time to drop old versions from the build matrix. Not doing so **will break** `gulpjs/glob-watcher` fsevents watching on nodejs 14+.

### LINKED ISSUE #58: Test suite consistently failing on Node.js 12 and 14
This fails with a variety of abort traps. It would appear that the source of the issue in the old version of chokidar

### LINKED ISSUE #64: fsevents not accessible from glob-watcher:chokidar in GitHub Actions
Before you open this issue, please complete the following tasks:

* [x] use the search bar at the top of the page to search this repository for similar issues or discussions that have already been opened.
* [x] if you are looking for help from the gulp team or community, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a problem with the plugin you're using, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a bug in our code, open this issue.

### What were you expecting to happen?

Hi, I'm using GitHub actions in my project. I expect `npm ci` to install my dependencies.

### What actually happened?

When running `npm ci` in GH Actions, I get the following error:

`npm ERR! fsevents not accessible from glob-watcher:chokidar`

See here: https://github.com/skaut/shared-drive-mover/runs/2272345131

### Please give us a sample of your gulpfile

I use gulp in the project, but the problem doesn't seem to be running it, just getting `npm ci` not to throw an error.

### Please provide the following information:
* OS & version [e.g. MacOS Catalina 10.15.4]: Ubuntu 20.04
* node version (run `node -v`): 14.16.0
* npm version (run `npm -v`): 6.14.11

All according to https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md

### Additional information
I'm not 100% sure the issue is caused by this package, but seems related to #55

### PR DESCRIPTION
**Pull request by sttk**
This PR updates this project:

- Updates dependency versions
- Updates GitHub Actions files
- Adds, changes and deletes some project files
- Drops `anymatch`, `is-negated-glob`, `just-debounce`, `normalize-path` and `object.defaults` from dependencies.
    - Removes glob opratons about positive/nagative because it is supported by latest `chokidar`.
    - Adds alternative code of `just-debounce` to `lib/debounce.js`'
- Move argments operations to `lib/normalize-args.js`

**NOTE:** Latest `chokidar` is seemed to [accept windows-style glob paths](https://github.com/gulpjs/glob-watcher/compare/master...sttk:glob-watcher:chore-normalize-repository?expand=1#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440bR25-R26), though  [`chokidar`'s `README.md`](https://github.com/paulmillr/chokidar#api) says *'globs must not contain windows separators'*.

Closes #61
Closes #60 
Closes #64
Closes #58
Closes #55
Closes #49

### CODE REVIEW

**Comment by iChenLei** [comment]
Maintainers, how is the progress of this PR currently? : )

**Comment by garohussenjian** [comment]
Also please consider this malware advisory: https://github.com/advisories/GHSA-xv2f-5jw4-v95m

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2017 Blaine Bublitz <blaine.bublitz@gmail.com>, Eric Schoffstall <yo@contra.io> and other contributors
+Copyright (c) 2017-2018, 2020, 2022 Blaine Bublitz <blaine.bublitz@gmail.com> and Eric Schoffstall <yo@contra.io>.
```
Comment (phated) [inline LICENSE:3]:
```suggestion
Copyright (c) 2017-2018, 2020, 2022-2023 Blaine Bublitz <blaine.bublitz@gmail.com> and Eric Schoffstall <yo@contra.io>
```

**Review by phated — APPROVED** [review]
LGTM!

**Pull request merged**